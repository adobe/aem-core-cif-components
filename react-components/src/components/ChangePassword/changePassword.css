/*******************************************************************************
 *
 *    Copyright 2021 Adobe. All rights reserved.
 *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License. You may obtain a copy
 *    of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software distributed under
 *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 *    OF ANY KIND, either express or implied. See the License for the specific language
 *    governing permissions and limitations under the License.
 *
 ******************************************************************************/
.root {
    display: grid;
    gap: 1rem;
    padding: 1.5rem;
}

.message {
    background-color: rgb(var(--venia-grey));
    border-radius: 4px;
    font-size: 0.875rem;
    line-height: 1.25rem;
    padding: 1rem;
}

.message:empty {
    display: none;
}

.actions {
    display: flex;
    justify-content: space-around;
    margin-top: 1rem;
}

.error {
    composes: message;
    color: rgb(var(--venia-error));
}

.lead {
    composes: message;
}
