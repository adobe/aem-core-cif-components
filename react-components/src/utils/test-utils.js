/*******************************************************************************
 *
 *    Copyright 2019 Adobe. All rights reserved.
 *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License. You may obtain a copy
 *    of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software distributed under
 *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 *    OF ANY KIND, either express or implied. See the License for the specific language
 *    governing permissions and limitations under the License.
 *
 ******************************************************************************/
import React from 'react';
import { render } from '@testing-library/react';
import { MockedProvider } from '@apollo/client/testing';
import { I18nextProvider } from 'react-i18next';

import i18n from '../../__mocks__/i18nForTests';
import ConfigContextProvider from '../context/ConfigContext';
import UserContextProvider from '../context/UserContext';

import mutationChangePassword from './mocks/mutationChangePassword';
import mutationCreateCustomer from './mocks/mutationCreateCustomer';
import mutationCreateDuplicateCustomer from './mocks/mutationCreateCustomerDuplicate';
import mutationDeleteCustomerAddress from './mocks/mutationDeleteCustomerAddress';
import mutationGenerateToken from './mocks/mutationGenerateToken';
import mutationGenerateTokenWrongPassword from './mocks/mutationGenerateTokenWrongPassword';
import mutationMergeCarts from './mocks/mutationMergeCarts';
import mutationPlaceOrder from './mocks/mutationPlaceOrder';
import mutationRevokeToken from './mocks/mutationRevokeToken';
import mutationShippingAddress from './mocks/mutationShippingAddress';
import queryCart from './mocks/queryCart';
import queryCountries from './mocks/queryCountries';
import queryCustomerCart from './mocks/queryCustomerCart';
import queryCustomerDetails from './mocks/queryCustomerDetails';
import queryCustomerInformation from './mocks/queryCustomerInformation';
import queryEmptyCart from './mocks/queryEmptyCart';
import queryNewCart from './mocks/queryNewCart';

const defaultMocks = [
    mutationChangePassword,
    mutationCreateCustomer,
    mutationCreateDuplicateCustomer,
    mutationDeleteCustomerAddress,
    mutationGenerateToken,
    mutationGenerateTokenWrongPassword,
    mutationMergeCarts,
    mutationPlaceOrder,
    mutationRevokeToken,
    mutationShippingAddress,
    queryCart,
    queryCountries,
    queryCustomerCart,
    queryCustomerDetails,
    queryCustomerInformation,
    queryEmptyCart,
    queryNewCart
];

const defaultConfig = {
    storeView: 'default',
    graphqlEndpoint: 'none'
};

// eslint-disable-next-line react/display-name
const allProviders = (config, userContext, mocks) => ({ children }) => {
    return (
        <MockedProvider mocks={mocks} addTypename={false}>
            <ConfigContextProvider config={config || defaultConfig}>
                <UserContextProvider initialState={userContext}>
                    <I18nextProvider i18n={i18n}>{children}</I18nextProvider>
                </UserContextProvider>
            </ConfigContextProvider>
        </MockedProvider>
    );
};

/* Wrap all the React components tested with the library in a mocked Apollo provider */
const customRender = (ui, options = {}) => {
    const { config, userContext, mocks = defaultMocks, ...renderOptions } = options;
    return render(ui, { wrapper: allProviders(config, userContext, mocks), ...renderOptions });
};

export * from '@testing-library/react';
export { customRender as render };
