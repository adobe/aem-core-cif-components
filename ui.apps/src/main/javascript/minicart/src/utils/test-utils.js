/*******************************************************************************
 *
 *    Copyright 2019 Adobe. All rights reserved.
 *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License. You may obtain a copy
 *    of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software distributed under
 *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 *    OF ANY KIND, either express or implied. See the License for the specific language
 *    governing permissions and limitations under the License.
 *
 ******************************************************************************/

import React from 'react';
import { render } from '@testing-library/react';
import { MockedProvider } from '@apollo/react-testing';

import QUERY_CART_DETAILS from '../queries/query_cart_details.graphql';

const emptyCartId = 'empty';
const mockCartId = '123ABC';

const mocks = [
    {
        request: {
            query: QUERY_CART_DETAILS,
            variables: {
                cartId: emptyCartId
            }
        },
        result: {
            data: {
                cart: {
                    email: null,
                    shipping_addresses: [],
                    prices: {
                        grand_total: {
                            currency: 'USD',
                            value: 0
                        }
                    },
                    selected_payment_method: {
                        code: '',
                        title: ''
                    },
                    billing_address: {
                        city: null,
                        country: {
                            code: null
                        },
                        lastname: null,
                        firstname: null,
                        region: {
                            code: null
                        },
                        street: [''],
                        postcode: null,
                        telephone: null
                    },
                    available_payment_methods: [
                        {
                            code: 'cashondelivery',
                            title: 'Cash On Delivery'
                        },
                        {
                            code: 'banktransfer',
                            title: 'Bank Transfer Payment'
                        },
                        {
                            code: 'checkmo',
                            title: 'Check / Money order'
                        },
                        {
                            code: 'free',
                            title: 'No Payment Information Required'
                        }
                    ],
                    items: []
                }
            }
        }
    },
    {
        request: {
            query: QUERY_CART_DETAILS,
            variables: {
                cartId: mockCartId
            }
        },
        result: {
            data: {
                prices: {
                    grand_total: {
                        currency: 'USD',
                        value: 0
                    }
                },
                items: []
            }
        }
    }
];

const AllProviders = ({ children }) => {
    return (
        <MockedProvider mocks={mocks} addTypename={false}>
            {children}
        </MockedProvider>
    );
};

/* Wrap all the React components tested with the library in a mocked Apollo provider */
const customRender = (ui, options) => render(ui, { wrapper: AllProviders, ...options });

export * from '@testing-library/react';
export { customRender as render };
