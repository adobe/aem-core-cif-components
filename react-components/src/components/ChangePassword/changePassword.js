/*******************************************************************************
 *
 *    Copyright 2019 Adobe. All rights reserved.
 *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License. You may obtain a copy
 *    of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software distributed under
 *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 *    OF ANY KIND, either express or implied. See the License for the specific language
 *    governing permissions and limitations under the License.
 *
 ******************************************************************************/
import React from 'react';
import { useTranslation } from 'react-i18next';
import { func } from 'prop-types';
import { Form } from 'informed';
import { useMutation } from '@apollo/client';

import classes from './changePassword.css';
import { useUserContext } from '../../context/UserContext';
import Field from '../Field';
import TextInput from '../TextInput';
import Button from '../Button';
import combine from '../../utils/combineValidators';
import { isRequired, validatePassword, validateConfirmPassword, hasLengthAtLeast } from '../../utils/formValidators';
import parseError from '../../utils/parseError';
import LoadingIndicator from '../LoadingIndicator';

import MUTATION_CHANGE_PASSWORD from '../../queries/mutation_change_password.graphql';

const ChangePassword = props => {
    const { showMyAccount, handleCancel } = props;
    const [{ token }] = useUserContext();
    const [doChangePassword, { data, loading, error }] = useMutation(MUTATION_CHANGE_PASSWORD);
    const [t] = useTranslation(['account', 'common']);

    const handleSubmit = formValues => {
        doChangePassword({
            variables: {
                currentPassword: formValues.oldPassword,
                newPassword: formValues.password
            },
            context: {
                headers: {
                    authorization: `Bearer ${token && token.length > 0 ? token : ''}`
                }
            }
        });
    };

    if (loading) {
        return (
            <div className={classes.root}>
                <LoadingIndicator>Loading</LoadingIndicator>
            </div>
        );
    }

    if (data) {
        return (
            <div className={classes.root}>
                <p>{t('account:password-was-changed', 'Your password was changed.')}</p>
                <div className={classes.actions}>
                    <Button priority="high" onClick={showMyAccount}>
                        {t('common:back', 'Back')}
                    </Button>
                </div>
            </div>
        );
    }

    if (!data) {
        return (
            <div className={classes.root}>
                <Form onSubmit={handleSubmit}>
                    <Field label={t('account:current-password', 'Current Password')} required={true}>
                        <TextInput
                            field="oldPassword"
                            type="password"
                            validate={combine([isRequired])}
                            validateOnBlur
                            aria-label="old-password"
                        />
                    </Field>
                    <Field label={t('account:new-password', 'New Password')} required={true}>
                        <TextInput
                            field="password"
                            type="password"
                            autoComplete="password"
                            validate={combine([isRequired, [hasLengthAtLeast, 8], validatePassword])}
                            validateOnBlur
                            aria-label="password"
                        />
                    </Field>
                    <Field label={t('account:confirm-new-password', 'Confirm New Password')} required={true}>
                        <TextInput
                            field="confirm"
                            type="password"
                            validate={combine([isRequired, validateConfirmPassword])}
                            validateOnBlur
                            aria-label="confirm"
                        />
                    </Field>
                    {error && <div className={classes.error}>{parseError(error)}</div>}
                    <div className={classes.actions}>
                        <Button type="submit" priority="high" aria-label="submit">
                            {t('account:change-password', 'Change Password')}
                        </Button>
                        {handleCancel && (
                            <Button type="button" priority="normal" aria-label="cancel" onClick={handleCancel}>
                                {t('account:change-password-cancel', 'Cancel')}
                            </Button>
                        )}
                    </div>
                </Form>
            </div>
        );
    }
};

ChangePassword.propTypes = {
    showMyAccount: func.isRequired,
    handleCancel: func
};

export default ChangePassword;
