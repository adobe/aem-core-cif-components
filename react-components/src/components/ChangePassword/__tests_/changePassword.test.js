/*******************************************************************************
 *
 *    Copyright 2019 Adobe. All rights reserved.
 *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License. You may obtain a copy
 *    of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software distributed under
 *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 *    OF ANY KIND, either express or implied. See the License for the specific language
 *    governing permissions and limitations under the License.
 *
 ******************************************************************************/

import React from 'react';
import { fireEvent, waitForElement } from '@testing-library/react';
import { render } from 'test-utils';
import ChangePassword from '../changePassword';
import MUTATION_CHANGE_PASSWORD from '../../../queries/mutation_change_password.graphql';

describe('<ChangePassword />', () => {
    it('renders the change password form', () => {
        const { asFragment } = render(<ChangePassword showMyAccount={() => {}} />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('renders the success message', async () => {
        const mocks = [
            {
                request: {
                    query: MUTATION_CHANGE_PASSWORD,
                    variables: {
                        currentPassword: 'old-password',
                        newPassword: 'NewPassword123'
                    }
                },
                result: {
                    data: {
                        changeCustomerPassword: {
                            id: 4
                        }
                    }
                }
            }
        ];

        const { getByLabelText, getByText } = render(<ChangePassword showMyAccount={() => {}} />, { mocks: mocks });

        const detailsFromValue = value => {
            return {
                target: {
                    value
                }
            };
        };
        fireEvent.change(getByLabelText('old-password'), detailsFromValue('old-password'));
        fireEvent.change(getByLabelText('password'), detailsFromValue('NewPassword123'));
        fireEvent.change(getByLabelText('confirm'), detailsFromValue('NewPassword123'));
        fireEvent.click(getByLabelText('submit'));

        const result = await waitForElement(() => {
            return getByText('Your password was changed.');
        });

        expect(result).not.toBeUndefined();
    });
});
