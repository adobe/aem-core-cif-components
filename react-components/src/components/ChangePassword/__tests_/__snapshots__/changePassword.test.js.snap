// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChangePassword /> renders the change password form 1`] = `
<DocumentFragment>
  <div
    class="root"
  >
    <form>
      <div
        class="root"
      >
        <span
          class="label"
        >
          <span
            class="requiredSymbol"
          />
          Current Password
        </span>
        <span
          class="root"
        >
          <span
            class="input"
          >
            <input
              aria-label="old-password"
              class="input"
              name="oldPassword"
              type="password"
              value=""
            />
          </span>
          <span
            class="before"
          />
          <span
            class="after"
          />
        </span>
        <p
          class="root"
        />
      </div>
      <div
        class="root"
      >
        <span
          class="label"
        >
          <span
            class="requiredSymbol"
          />
          New Password
        </span>
        <span
          class="root"
        >
          <span
            class="input"
          >
            <input
              aria-label="password"
              autocomplete="password"
              class="input"
              name="password"
              type="password"
              value=""
            />
          </span>
          <span
            class="before"
          />
          <span
            class="after"
          />
        </span>
        <p
          class="root"
        />
      </div>
      <div
        class="root"
      >
        <span
          class="label"
        >
          <span
            class="requiredSymbol"
          />
          Confirm New Password
        </span>
        <span
          class="root"
        >
          <span
            class="input"
          >
            <input
              aria-label="confirm"
              class="input"
              name="confirm"
              type="password"
              value=""
            />
          </span>
          <span
            class="before"
          />
          <span
            class="after"
          />
        </span>
        <p
          class="root"
        />
      </div>
      <div
        class="actions"
      >
        <button
          aria-label="submit"
          class="root_highPriority"
          type="submit"
        >
          <span
            class="content"
          >
            Change Password
          </span>
        </button>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
